sdk/eventhub/event-hubs/src/eventProcessor.ts

```diff
@@ -476,6 +446,66 @@ export class EventProcessor {
     this._isRunning = false;
   }
 
+  private async _performLoadBalancing(
+    loadBalancingStrategy: LoadBalancingStrategy,
+    partitionIds: string[],
+    abortSignal: AbortSignalLike
+  ) {
+    if (abortSignal.aborted) return;
+
+    // Retrieve current partition ownership details from the datastore.
+    const partitionOwnership = await this._checkpointStore.listOwnership(
+      this._fullyQualifiedNamespace,
+      this._eventHubName,
+      this._consumerGroup
+    );
+
+    if (abortSignal.aborted) return;
+
+    const partitionOwnershipMap = new Map<string, PartitionOwnership>();
+    const nonAbandonedPartitionOwnershipMap = new Map<string, PartitionOwnership>();
+    const partitionsToRenew: string[] = [];
+
+    // Separate abandoned ownerships from claimed ownerships.
+    // We only want to pass active partition ownerships to the
+    // load balancer, but we need to hold onto the abandoned
+    // partition ownerships because we need the etag to claim them.
+    for (const ownership of partitionOwnership) {
+      partitionOwnershipMap.set(ownership.partitionId, ownership);
+      if (!isAbandoned(ownership)) {
+        nonAbandonedPartitionOwnershipMap.set(ownership.partitionId, ownership);
+      }
+      if (
+        ownership.ownerId === this._id &&
+        this._pumpManager.isReceivingFromPartition(ownership.partitionId)
+      ) {
+        partitionsToRenew.push(ownership.partitionId);
+      }
+    }
+
+    // Pass the list of all the partition ids and the collection of claimed partition ownerships
+    // to the load balance strategy.
+    // We exclude the abandoned partition ownerships to simplify the load balancing logic.
```

Comment:
I'm not sure if I get why we'd exclude abandoned partitions? Why aren't they considered in load balancing?

Reply:
The load balancing strategy expects a list of current partition ownerships. Abandoned partitions are considered as not being owned (not current), so they are excluded from the map.

I could move this logic into the load balancing algorithm instead (though I do kind of like that it's one less thing the load balancing strategy has to think about :smile: )

Reply:
I improved the comments here to hopefully better explain why we do this.

Reply:
(having conversation offline)